loader/CMakeLists.txt

```diff
@@ -216,22 +217,32 @@ elseif(APPLE)
 else(UNIX AND NOT APPLE) # i.e.: Linux
     enable_language(ASM)
     set(CMAKE_ASM_FLAGS "${CMAKE_C_FLAGS}")
+    set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)
+
+    if (${CMAKE_HOST_SYSTEM_PROCESSOR} STREQUAL "aarch64")
```

Comment:
The change doesn't work correctly when cross-compiling loader from x86_64 platform with aarch64-linux-gnu-.  There are two problems:
  The CMAKE_HOST_SYSTEM_PROCESSOR is x86_64 in that case. 
  When I work-arounded the first issue, I found asm_offset is also for aarch, so it cannot be execute on x86_64.

I think we have to address the two problems.

Reply:
Thanks for the feedback! Cross compiling was something I originally wanted to be part of this PR but I think it would be significant enough that it could justify its own. I do agree that it should be something that is supported eventually though. I would recommend we review this PR keeping in mind it currently requires the aarch64 loader with the assembly unknown extension implementation to be built on an aarch64 system. I could add a flag that would force the use of the C implementation for cross compiling though. We could then discuss options on how to support cross compiling with the aarch64 assembly unknown extension implementation in a separate issue. I do have some ideas on how to remove the asm_offset executable while still having the compiler generate the struct offsets. I am not sure they are any good though.

Reply:
No problem. We could implemented the cross-compile in a separate PR. 
And I added a new comment to this PR, please have a look. I think we don't know if those general-purpose registers Xn (in your implementation, they are X9, X10, X11) are already used by compiler or not, a typical example is that X9 will be used to store the 10th parameter if the method has more than 10 parameters.  So we need to store/recovery them. Do you agree?

Reply:
I have updated the PR to add an option to disable assembly sources. Thanks again for taking the time to provide some feedback.

Reply:
I noticed the cross-compiling issue while fixing #783, so I think it's now fixed as part of #784.